Adds Documentation for Config Service and adapts Configuration Wiki (#84)
eclipse-basyx/basyx-wiki: docs/source/content/developer_documentation/basyx_go/configuration_service/index.md

# BaSyx Configuration Service Developer Documentation

This documentation describes the internal architecture and extension points of the BaSyx Configuration Service.

> The service is implemented as a small ordered sequence runner. It connects to PostgreSQL, prepares the BaSyx schema-version table, uploads the base schema when needed, and applies explicitly registered SQL patches.

Reviewer: ```suggestion
The service is implemented as a small ordered sequence runner. It connects to PostgreSQL, creates and preseeds the BaSyx schema-version table, uploads the base schema if not present, and applies registered SQL patches.
```
